doc/ir-semantics.text

```diff
@@ -152,13 +149,12 @@ Update (partial) function, returns an H:
 ## One step reduction
 
   step   (const x = e) (P, I, T, H, E, l)
-       = (I, T, H, E[x ↦ v], succ I l)
-    when v := eval P H E e
+       = (I, T, H+H', E+E', succ I l)
+   where (H', E') := create-env H E (const x = e)
```

Comment:
Ugh, I much rather prefer duplication than reusing something in unrelated contexts. This feels a bit forced to me... (similar to inheriting from completely unrelated superclass to acquire one useful method). To some extend I guess it's a matter of taste. But I may add it's brittle since the fact that declaration and argument creation can use the same api is somewhat accidental.